Lib/test/test_pty.py

```diff
@@ -60,11 +65,26 @@ def _readline(fd):
     reader = io.FileIO(fd, mode='rb', closefd=False)
     return reader.readline()
 
+def expectedFailureIfStdinIsTTY(fun):
+    # avoid isatty() for now
+    try:
+        tty.tcgetattr(pty.STDIN_FILENO)
+        return unittest.expectedFailure(fun)
+    except tty.error:
+        pass
+    return fun
+
+def expectedFailureOnBSD(fun):
+    PLATFORM = platform.system()
+    if PLATFORM.endswith("BSD") or PLATFORM == "Darwin":
+        return unittest.expectedFailure(fun)
+    return fun
 
 
 # Marginal testing of pty suite. Cannot do extensive 'do or fail' testing
 # because pty code is not too portable.
 # XXX(nnorwitz):  these tests leak fds when there is an error.
+# Soumendra: test_openpty may leave tty in abnormal state upon failure.
```

Comment:
Could we potentially surround with a `try...finally` to avoid this? A single test failure should really not result in the other tests within `test_pty` being unreliable.

Reply:
Good point. Thank you. I will work on this.